File: java/interview/src/main/java/com/d00216118/demo/repository/UrlRepository.java
package com.d00216118.demo.repository;

import com.d00216118.demo.model.InfoUrl;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * @author Yu Chen
 * @email D00216118@student.dkit.ie
 * @date 1:12 下午 2021/4/2
 **/
@Repository
public interface UrlRepository extends JpaRepository<InfoUrl, Long> {

    InfoUrl findByUrl(String url);

    List<InfoUrl> findByMd5Url(String url);

    InfoUrl findByTinyUrlAndUserId(String tinyUrl,Long userId);

}

File: java/src/main/java/com/example/urlshortener/repository/UrlRepository.kt
package com.example.urlshortener.repository

import com.example.urlshortener.model.ShortUrl
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import javax.transaction.Transactional

/**
 * @author  lilu - Li Lu (7939743@qq.com)
 * @version 1.0
 * @since   2022/04/19
 */
@Repository
@Transactional
interface UrlRepository : JpaRepository<ShortUrl, String>